.github/workflows/site-deployment.yml

```
@@ -3,15 +3,20 @@ name: Site deployment
 on:
   deployment_status:
   schedule:
-    - cron: "*/10 * * * *"
+    - cron: "*/10 * * * *" ## Every 10 minutes
+    - cron: "0 4 * * *" ## Every night (UTC time)
+
+env:
+  PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: true
 
 jobs:
   warm_up_vercel:
     ## Vercel uses AWS Lambda. When we call getStaticProps in fallback mode or getServerSideProps,
     ## a cold Lambda function may take 5-15 seconds to execute. This pipeline warms up Lambda
     ## containers for new deployments and keeps them up by fetching affected URLs every 10 minutes.
+    ## Context: https://github.com/orgs/vercel/discussions/496
     name: Warm up Vercel
-    if: github.event_name == 'schedule' || github.event.deployment_status.state == 'success'
+    if: github.event.deployment_status.state == 'success' || github.event_name == 'schedule'
```

Not sure if this would fire twice at 4am using the previous approach, but probably worth tying it to its intended schedule in any case, so we don't have to worry about it if we introduce more.
```suggestion
    if: github.event.deployment_status.state == 'success' || github.event.schedule == '*/10 * * * *'
```

---

I’ve defined `needs: [warm_up_vercel]` in `smoke_tests` to run tasks in a sequence. If we change this condition, the `warm_up_vercel` will be skipped and so `smoke_tests` will probably not get triggered.

GitHub’s `cron` schedule is [quite inaccurate](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule), i.e. the runs can be postponed or even skipped. I see no harm in warming up Vercel functions ≈145 times instead of ≈144*

* ≈ once in 10 mins × 24 hours

---

I think that once in 5-10 mins is the maximum frequency GitHub can give: https://stackoverflow.com/a/63192190/1818285